### PR TITLE
Update SES nodemailer typings to match current implementation

### PR DESCRIPTION
**Pull request by bwindsor**
Please fill in this template.

- [ x ] Use a meaningful title for the pull request. Include the name of the package modified.
- [ x ] Test the change in your own code. (Compile and run.)
- [ x ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ x ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ x ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present). - no tslint.json present

Select one of these and delete the others:

If changing an existing definition:
- [ x ] Provide a URL to documentation or source code which provides context for the suggested changes: (see below)
- [ x  ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

**Changes**:
* Update SES options typings to match [this version](https://github.com/nodemailer/nodemailer/tree/master/lib/ses-transport)
* Add version of `createTransport` to nodemailer which accepts SES transport as options
* Remove `templateSender` from Transport type since this doesn't seem to exist any more

**Query**:
There seem to be two versions of `nodemailer-ses-transport` around, [one merged in with nodemailer](https://github.com/nodemailer/nodemailer/tree/master/lib/ses-transport) and [one separately](https://github.com/nodemailer/nodemailer-ses-transport). The separate one seems to be what the typings are based on, but the combined one seems to have been updated more recently.

First PR - typescript wasn't working for me with these packages so I thought I would try to fix it for everybody - any advice appreciated.

### CODE REVIEW

**Comment by RyanCavanaugh** [comment]
@bwindsor Please fix the failures indicated in the Travis CI log.

**Comment by bwindsor** [comment]
@RyanCavanaugh CI now passes.

**Comment by MR4online** [comment]
I'm getting the following Error after updating to Nodemailer 4.0.0 (and @types/nodemailer), which may be the Result of this merge:

node_modules/@types/nodemailer-ses-transport/index.d.ts(11,12): error TS2694: Namespace '"aws-sdk"' has no exported member 'SES'.

**Comment by andy-ms** [comment]
@MR4online I can't reproduce this just with `npm install @types/nodemailer` -- could you try to reproduce that in a sample project? Maybe you have an older `aws-sdk` installed?

File: types/nodemailer-ses-transport/index.d.ts [hunk 1]
```diff
@@ -1,22 +1,17 @@
-// Type definitions for nodemailer-ses-transport 1.4
+// Type definitions for nodemailer-ses-transport 3.1
```
Comment (pmoleri) [inline types/nodemailer-ses-transport/index.d.ts:1]: Hi. I just noticed that the new version doesn't match mayor/minor of any [published version of the package](https://www.npmjs.com/package/nodemailer-ses-transport).

Is this a mistake?
Shouldn't we downgrade it to 1.5?

Comment (pmoleri) [reply types/nodemailer-ses-transport/index.d.ts:1]: I created [PR 51291](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/51291) to fix this.

**Pull request merged**